src/cargo/util/config/mod.rs

```diff
@@ -742,9 +742,20 @@ impl Config {
                 .unwrap_or(false);
         self.target_dir = cli_target_dir;
 
+        // If nightly features are enabled, allow setting Z-flags from config
+        // using the `unstable` table. Ignore that block otherwise.
         if nightly_features_allowed() {
-            if let Some(val) = self.get::<Option<bool>>("unstable.mtime_on_use")? {
-                self.unstable_flags.mtime_on_use |= val;
+            if let Some(unstable_configs) =
+                self.get::<Option<HashMap<String, String>>>("unstable")?
```

Comment:
A worry I have about a string=>string table is that it can be a bit surprising:

* For `build-std` I'd expect to be able to do `build-std = ['panic-abort', 'std']`
* For boolean keys I'd expect `foo = true`

Requiring a string feels a bit odd compared to what Cargo idiomatically has as configuration.

Reply:
Another downside is that by using a `HashMap` env vars don't work. Otherwise you could do `CARGO_UNSTABLE_MTIME_ON_USE=true` for configuring Cargo.

Reply:
Agreed. I much prefer getting env along for the ride as well — I'll see about making the necessary changes to move this to serde, it just might end up being a little bit more of a refactor.